aibolit/patterns/er_class/er_class.py

```diff
@@ -23,5 +24,14 @@ def value(self, filename: str):
                    'producer',
                    'holder',
                    'interceptor')
-        tree = build_ast(filename).filter(javalang.tree.ClassDeclaration)
-        return [node._position.line for _, node in tree if [n for n in classes if n in node.name.lower()] != []]
+        ast = AST.build_from_javalang(build_ast(filename))
+        class_decls = list(ast.get_proxy_nodes(ASTNodeType.CLASS_DECLARATION))
+        for node in class_decls:
```

Comment:
```suggestion
        for node in ast.get_proxy_nodes(ASTNodeType.CLASS_DECLARATION):
```
And remove the line above. This way we would not create list of all nodes, but iterate one by one, without storing them all in memory.

Reply:
done